---
string[] predicates
string[] objects
string[] init
PredicateList predicateList
